File: 180-lab8/sort.py
def bubbleSort(array):
   if type(array) != list:
      return [False, array]
   if len(array) == 0:
      return [False, array]
   for i in range(0,len(array),1):
      if type(array[i]) != int and type(array[i]) != float:
         return [False, array]

   arraymod=list(array)
   n=len(arraymod)
   swapped = True
   
   while swapped == True:
      swapped = False
      for i in range(1,n,1):
         if arraymod[i-1] > arraymod[i]:
            temp = arraymod[i]
            arraymod[i]=arraymod[i-1]
            arraymod[i-1]=temp
            swapped = True

   return[True, arraymod]

File: 180-lab8/sort.c
#include <stdio.h>
int bubbleSort(int *array, int size);

int bubbleSort(int *array, int size){
   int swapped = 0;
   int i = 0;

   if (size <= 0){
      return -1;
   }else if (array == NULL){
      return -1;
   }

   while(swapped == 0){
      swapped = -1;
      for (i=1;i<size;i++){
         if (array[i-1] > array[i]){
            int temp = 0;
            temp=array[i];
            array[i]=array[i-1];
            array[i-1]=temp;
            swapped = 0;        
         }
      }
   }
   return 0;
   
}

/*
int main(void){
   int array[] = {1,4,5,6,8,7,3};
   int r = 0;
   int i = 0;
   array[4]=1000;
   r = bubbleSort(array,7);
   
   for (i=0;i<7;i++){
   printf("%d ", array[i]);
   }
}
*/
